azure-proto-compute/Extensions/AzureSubscriptionExtension.cs

```diff
@@ -15,31 +17,44 @@ public static class AzureSubscriptionExtension
         /// List vms at the given subscription context
         /// </summary>
         /// <param name="subscription"></param>
-        /// <param name="filter"></param>
-        /// <param name="top"></param>
-        /// <param name="cancellationToken"></param>
         /// <returns></returns>
-        public static Pageable<VirtualMachine> ListVirtualMachines(this SubscriptionOperations subscription, ArmSubstringFilter filter = null, int? top = null, CancellationToken cancellationToken = default)
+        public static Pageable<VirtualMachine> ListVirtualMachines(this SubscriptionOperations subscription)
         {
-            //TODO: consider ArmPageable<T> to introduce post network call filtering and avoid breaking changes
-            ArmFilterCollection filters = new ArmFilterCollection(VirtualMachineData.ResourceType);
-            filters.SubstringFilter = filter;
-            return ResourceListOperations.ListAtContext<VirtualMachine, VirtualMachineData>(subscription, filters, top, cancellationToken);
+            var instance = subscription.GetClient((baseUri, cred) => new ComputeManagementClient(baseUri, subscription.Id.Subscription, cred,
```

Comment:
rename variable to vmOperations
```suggestion
            var vmOperations = subscription.GetClient((baseUri, cred) => new ComputeManagementClient(baseUri, subscription.Id.Subscription, cred,
```